src/backend/main.ts

```diff
@@ -1007,227 +1064,255 @@ ipcMain.handle(
         `Game launched at: ${startPlayingDate}\n` +
         '\n'
     )
-    return game
-      .launch(launchArguments)
-      .catch((exception) => {
-        logError(exception, { prefix: LogPrefix.Backend })
+
+    let launchResult = false
+    try {
+      launchResult = await game.launch(launchArguments)
+    } catch (error) {
+      logError(error, { prefix: LogPrefix.Backend })
+      if (error instanceof Error) {
         appendFileSync(
           game.logFileLocation,
-          `An exception occurred when launching the game:\n${exception.stack}`
+          `An exception occurred when launching the game:\n${error.stack}`
         )
+      }
+    }
+
+    // Stop display sleep blocker
+    if (powerDisplayId !== null) {
+      logInfo('Stopping Display Power Saver Blocker', {
+        prefix: LogPrefix.Backend
       })
-      .finally(() => {
-        // Stop display sleep blocker
-        if (powerDisplayId !== null) {
-          logInfo('Stopping Display Power Saver Blocker', {
-            prefix: LogPrefix.Backend
-          })
-          powerSaveBlocker.stop(powerDisplayId)
-        }
+      powerSaveBlocker.stop(powerDisplayId)
+    }
 
-        // Update playtime and last played date
-        const finishedPlayingDate = new Date()
-        tsStore.set(`${game.appName}.lastPlayed`, finishedPlayingDate)
-        // Playtime of this session in minutes
-        const sessionPlaytime =
-          (finishedPlayingDate.getTime() - startPlayingDate.getTime()) /
-          1000 /
-          60
-        let totalPlaytime = sessionPlaytime
-        if (tsStore.has(`${game.appName}.totalPlayed`)) {
-          totalPlaytime += tsStore.get(`${game.appName}.totalPlayed`) as number
-        }
-        tsStore.set(`${game.appName}.totalPlayed`, Math.floor(totalPlaytime))
+    // Update playtime and last played date
+    const finishedPlayingDate = new Date()
+    tsStore.set(`${game.appName}.lastPlayed`, finishedPlayingDate)
+    // Playtime of this session in minutes
+    const sessionPlaytime =
+      (finishedPlayingDate.getTime() - startPlayingDate.getTime()) / 1000 / 60
+    let totalPlaytime = sessionPlaytime
+    if (tsStore.has(`${game.appName}.totalPlayed`)) {
+      totalPlaytime += tsStore.get(`${game.appName}.totalPlayed`) as number
+    }
+    tsStore.set(`${game.appName}.totalPlayed`, Math.floor(totalPlaytime))
 
-        window.webContents.send('setGameStatus', {
-          appName,
-          runner,
-          status: 'done'
-        })
+    window.webContents.send('setGameStatus', {
+      appName,
+      runner,
+      status: 'done'
+    })
 
-        // Exit if we've been launched without UI
-        if (isCLINoGui) {
-          app.exit()
-        } else {
-          mainWindow.show()
-        }
-      })
+    // Show main window if we've minimized it before
+    if (minimizeOnLaunch) {
+      mainWindow.show()
+    }
```

Comment:
do we need this separation of the if/else? this is checking the `minimizeOnLaunch` setting. I think that if we launch a game with the `no-gui` flag and we have the minimizeOnLaunch flag turned on, this 2 ifs will first show the app and then close it, while the original if/else is either closing the app or showing it.

Reply:
Hm, I see the issue. The reason I've changed this `if` was that Heroic would take focus after a game quit even if we weren't minimized/hidden  
I'll just return it to how it was before, not sure why I've added this change here anyways